storage/index/options.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package index
+
+import (
+	"errors"
+
+	"github.com/m3db/m3db/clock"
+	"github.com/m3db/m3db/x/xpool"
+	"github.com/m3db/m3ninx/index/segment/mem"
+	"github.com/m3db/m3x/ident"
+	"github.com/m3db/m3x/instrument"
+)
+
+const (
+	defaultCheckedBytesPoolInitialSize = 128
+)
+
+var (
+	errOptionsIdentifierPoolUnspecified = errors.New("identifier pool is unset")
+	errOptionsWrapperPoolUnspecified    = errors.New("checkedbyteswrapper pool is unset")
+)
+
+type opts struct {
+	insertMode     InsertMode
+	clockOpts      clock.Options
+	instrumentOpts instrument.Options
+	memOpts        mem.Options
+	idPool         ident.Pool
+	wrapperPool    xpool.CheckedBytesWrapperPool
+}
+
+// NewOptions returns a new index.Options object with default properties.
+func NewOptions() Options {
+	idPool := ident.NewPool(nil, nil)
+	wrapperPool := xpool.NewCheckedBytesWrapperPool(defaultCheckedBytesPoolInitialSize)
+	wrapperPool.Init()
+	return &opts{
+		insertMode:     InsertAsync,
```

Comment:
Hm I believe that for defaults for series inserts it is synchronous, we should match the defaults of that with this - if writes are sync by default, so too should be indexing yeah?

Reply:
Also can you make this a default const at the top of this file so it's more clear. i.e. `defaultInsertMode = XYZ`

Reply:
sure, will do